internal/oci/notation.go

```diff
@@ -0,0 +1,225 @@
+package oci
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/notaryproject/notation-go/dir"
+	verifier "github.com/notaryproject/notation-go/verifier"
+	"github.com/notaryproject/notation-go/verifier/truststore"
+
+	_ "github.com/notaryproject/notation-core-go/signature/cose"
+	_ "github.com/notaryproject/notation-core-go/signature/jws"
+	"github.com/notaryproject/notation-go"
+	"github.com/notaryproject/notation-go/registry"
+	"github.com/notaryproject/notation-go/verifier/trustpolicy"
+	oras "oras.land/oras-go/v2/registry/remote"
+	oauth "oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// notationOptions is a struct that holds options for notation verifier
+type notationOptions struct {
```

Comment:
I think we can just use the defined `options struct` in `verifier.go`. That would give us `func NewNotaryVerifier(opts ...Options) (*NotaryVerifier, error)`.

Reply:
Excellent idea! Resolved in [d068a41](https://github.com/fluxcd/source-controller/pull/1075/commits/d068a413d17ef2461f796dbfe1cd174ff915c84d).